### PR TITLE
RecommenderConfiguration and the Abstract RecommenderSystem and Conte…

### PR DESCRIPTION
**Pull request by mostafa-mahmoud**
…ntBased. Added tests for all of them.

### CODE REVIEW

**Comment by mostafa-mahmoud** [comment]
I need a function from the DataParser to get the documents x bag of words matrix.

File: config/recommender.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "recommender": {
+        "content-based": "LDA",
+        "collaborative-filtering": "ALS",
```
Comment (ndomar) [inline config/recommender.json]: ALS is the only method we are gonna use anyways, I think it is useless to put it in the configuration.

File: config/recommender.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    "recommender": {
+        "content-based": "LDA",
```
Comment (ndomar) [inline config/recommender.json]: Can we list all possible inputs for every field in an enum or something?

File: lib/content_based.py [hunk 3]
```diff
@@ -1,29 +1,31 @@
...
         """
         Constructor of ContentBased processor.
         """
-        pass
+        self.n_factors = n_factors
+        self.n_items = ratings.shape[0]
+        self.ratings = ratings
 
-    def process(self):
+    def train(self, num_iterations=10):
```
Comment (ndomar) [inline lib/content_based.py]: I think num_iterations should be loaded from the configuration file anyways.

File: lib/recommender_system.py [hunk 4]
```diff
@@ -1,25 +1,47 @@
...
         """
         Constructor of the RecommenderSystem.
         """
-        pass
+        # DataParser.process()
+        ratings = numpy.matrix(DataParser.get_ratings_matrix())
+        self.config = RecommenderConfiguration()
+        if self.config.get_content_based() == 'LDA':
+            pass
+        if self.config.get_content_based() == 'LDA2Vec':
```
Comment (ndomar) [inline lib/recommender_system.py]: If there is an error in the configuration file this won't run. if the input is LD for example. let's use a default method or simply quit if there is a mistake in the input

**Pull request merged**